airflow/ui/src/queries/useAddVariable.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useQueryClient } from "@tanstack/react-query";
+import { useState } from "react";
+
+import {
+  useVariableServiceGetVariablesKey,
+  useVariableServicePostVariable,
+} from "openapi/queries";
+import { toaster } from "src/components/ui";
+import type { AddVariableBody } from "src/pages/Variables/AddVariableForm";
+
+export const useAddVariable = (onClose: () => void) => {
+  const queryClient = useQueryClient();
+  const [error, setError] = useState<unknown>(undefined);
+
+  const onSuccess = async () => {
+    await queryClient.invalidateQueries({
+      queryKey: [useVariableServiceGetVariablesKey],
+    });
+
+    toaster.create({
+      description: "Variable has been added successfully",
+      title: "Variable Add Request Submitted",
+      type: "success",
+    });
+
+    onClose();
+  };
+
+  const onError = (_error: unknown) => {
+    setError(_error);
+  };
+
+  const { isPending, mutate } = useVariableServicePostVariable({
+    onError,
+    onSuccess,
+  });
+
+  const addVariable = (addVariableRequestBody: AddVariableBody) => {
+    mutate({
```

Comment:
If I add a variable which is already existing it is silently overwritten w/o showing an error. Can you please change the behavior such that adding is not silently overwriting an existing? Showing an error like the legacy might be a good option.

Reply:
Ohh okay, sure
1. Should we display the warning on submit and and pop up a checkbox in form to override?

2. or simple error with blocking user from having same name, make sense?

I think, 2 makes more sense, since patch is also available to edit the existing key, if the user really want to have same name, they can go and edit the variable.

What would you say? @jscheffl

Reply:
I would also propose to use (2) as the user might want to correct and re-submit. Else if you use (1) then you need to repeat all the add and fill the fields again.

Reply:
@bbovenzi @jscheffl 
Should this be handled from backend or from client side using calling getVariable API?
In the legacy we have the import variables options [which is a TODO], that have options to override, delete, so to get that feature in the new UI, do we need to implement it?

Reply:
On the backend.

Add variable shouldn’t behave as an “upsert” updating if the variable is already present. That should respond a 409 conflict.

Reply:
From point of code I assume this can be resolved?